src/oc_erchef/apps/oc_chef_wm/src/chef_wm_depsolver.erl

```diff
@@ -278,22 +281,65 @@ wm_halt(Code, Req, State, ErrorData, LogMsg) ->
 %% removing large fields such as long_description and attributes in
 %% the metadata that are not required by chef-client
 assemble_response(Req, #base_state{server_api_version = ApiVersion} = State, CookbookVersions) ->
+
     case oc_chef_wm_base:check_cookbook_authz(CookbookVersions, Req, State) of
         ok ->
-            %% We iterate over the list again since we only want to construct the s3urls
-            %% if the authz check has succeeded.  We use a minimal version of the
-            %% cookbook which has just enough information for chef-client to run
-            JsonList = {
-                    [ { CBV#chef_cookbook_version.name,
-                       chef_cookbook_version:minimal_cookbook_ejson(CBV, chef_wm_util:base_uri(Req), ApiVersion) }
-                      || CBV <- CookbookVersions ]
-                    },
-            CBMapJson = chef_json:encode(JsonList),
-            {true, wrq:append_to_response_body(CBMapJson, Req), State};
+            case make_json_list(CookbookVersions, chef_wm_util:base_uri(Req), ApiVersion) of
```

Comment:
Would it help if we reduce the number of calls to chef_cbv_cache.erl module in the case of 'cbv_cache_enabled = false'
Flag = envy:get(chef_objects, cbv_cache_enabled, false, boolean),
case make_json_list(CookbookVersions, chef_wm_util:base_uri(Req), ApiVersion, Flag) of

Function Defination:
make_json_list(CookbookVersions, URI, ApiVersion, false) ->
    get_json_list(CookbookVersions, URI, ApiVersion);
make_json_list(CookbookVersions, URI, ApiVersion, true) ->
    Hash = erlang:phash2(CookbookVersions, 134217728),
    make_json_list(CookbookVersions, URI, ApiVersion, Hash, 0).
    
get_json_list(CookbookVersions, URI, ApiVersion) ->
       {
                      [ { CBV#chef_cookbook_version.name,
                          chef_cookbook_version:minimal_cookbook_ejson(CBV, URI, ApiVersion) }
                        || CBV <- CookbookVersions ]
          }.

We can use get_json_list(CookbookVersions, URI, ApiVersion) at line 323.

Reply:
In my testing with service disabled, there was no significant impact to having the additional call in place -it failed at the same load that it failed without these changes in place.  Too, this way the chef_object configuration remains visible only within the module that needs it.